File: 1963-minimum-number-of-swaps-to-make-the-string-balanced/1963-minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character> st = new Stack<>();
        for (char c : s.toCharArray()) {
            if (c == '[') {
                st.push(c);
            } else if (!st.isEmpty()) {
                st.pop();
            }
        }
        return (st.size() + 1) / 2;
    }
}

File: LeetCode/1963-minimum-number-of-swaps-to-make-the-string-balanced/1963-minimum-number-of-swaps-to-make-the-string-balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        int size = 0;
        for (char &c : s) {
            if (c == '[') {
                size++;
            } else if (size != 0) {
                size--;
            }
        }
        return (size + 1) / 2;
    }
};







// class Solution {
// public:
//     int minSwaps(string s) {
//         stack<char> st;
//         for (char &ch : s) {
//             if (ch == '[') {
//                 st.push(ch);
//             } else if (!st.empty()) {
//                 st.pop();
//             }
//         }
//         return (st.size() + 1) / 2;
//     }
// };